scripts/generate_bom.dart

```diff
@@ -0,0 +1,227 @@
+// ignore_for_file: avoid_print
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:intl/intl.dart';
+import 'package:pubspec_parse/pubspec_parse.dart' as pubspec;
+
+import 'bom_analysis.dart';
+
+const packagesDir = 'packages';
+const versionsFile = 'VERSIONS.md';
+const versionsJsonFile = 'scripts/versions.json';
+const androidVersionFile =
+    '$packagesDir/firebase_core/firebase_core/android/gradle.properties';
+const iosVersionFile =
+    '$packagesDir/firebase_core/firebase_core/ios/firebase_sdk_version.rb';
+const webVersionFile =
+    '$packagesDir/firebase_core/firebase_core_web/lib/src/firebase_sdk_version.dart';
+const windowsVersionFile =
+    '$packagesDir/firebase_core/firebase_core/windows/CMakeLists.txt';
+
+// Used to display the packages in the correct order
+const List<String> packages = [
+  'firebase_core',
+  'firebase_auth',
+  'cloud_firestore',
+  'firebase_storage',
+  'cloud_functions',
+  'firebase_database',
+  'firebase_messaging',
+  'firebase_crashlytics',
+  'firebase_performance',
+  'firebase_remote_config',
+  'firebase_analytics',
+  'firebase_dynamic_links',
+  'firebase_in_app_messaging',
+  'firebase_app_check',
+  'firebase_app_installations',
+  'firebase_ml_model_downloader',
+];
+
+const jsonEncoder = JsonEncoder.withIndent('    ');
```

Comment:
```suggestion
const jsonEncoder = JsonEncoder.withIndent('  ');
```

nit: can we do 2 spaces for indentation like other source files in the repo